src/Azure.Functions.Cli/Common/DurableManager.cs

```diff
@@ -101,19 +106,51 @@ private void SetStorageServiceAndTaskHubClient(ref AzureStorageOrchestrationServ
             }
         }
 
+        private void Initialize(ref AzureStorageOrchestrationService orchestrationService, ref TaskHubClient taskHubClient, string connectionStringKey = null, string taskHubName = null)
+        {
+            CheckAssemblies();
+            SetStorageServiceAndTaskHubClient(ref _orchestrationService, ref _client, connectionStringKey, taskHubName);
+        }
+
+        private void CheckAssemblies()
+        {
+            // Retrieve list of DurableTask.AzureStorage DLL files
+            var assemblyFilePaths = Directory.GetFiles(Environment.CurrentDirectory, $"{DurableAzureStorageExtensionName}", SearchOption.AllDirectories).ToList<string>();
+            
+            if (assemblyFilePaths.Count() == 0)
+            {
+                ColoredConsole.WriteLine(Yellow($"Could not find {DurableAzureStorageExtensionName}. The functions host must be running a" +
+                    $" Durable Functions app in order for Durable Functions CLI commands to work."));
+                return;
+            }
+
+            // Convert them to a sorted list of FileInfo objects. Sort by the last modified time of the enclosing bin folder
+            var assemblyFiles = assemblyFilePaths.Select(assemblyFilePath => new FileInfo(assemblyFilePath)).ToList().OrderBy(file => file.Directory.LastWriteTime);
+
+            var minimumExtensionVersion = new Version(MinimumDurableAzureStorageExtensionVersion);
+
+            var mostRecentAssembly = FileVersionInfo.GetVersionInfo(assemblyFiles.LastOrDefault().FullName);
+            var mostRecentAssemblyVersion = new Version(mostRecentAssembly.FileVersion);
+
+            if (mostRecentAssemblyVersion < minimumExtensionVersion)
+            {
+                 throw new CliException($"Durable Functions CLI commands must be used with {DurableAzureStorageExtensionName} versions greater than or equal to 1.4.0." +
```

Comment:
Use your `MinimumDurableAzureStorageExtensionVersion` constant here.

Also, you should include the file path in this exception method in case that the CLI picks up a file that the user wasn't expecting it to pick up.